ee/clickhouse/queries/clickhouse_trends.py

```diff
@@ -1,10 +1,169 @@
-from typing import Any, Dict, List
+from datetime import datetime, timedelta, timezone
+from itertools import accumulate
+from typing import Any, Dict, List, Optional, Tuple
 
+from ee.clickhouse.client import ch_client
+from ee.clickhouse.models.action import format_action_filter
+from ee.clickhouse.models.property import parse_filter, parse_prop_clauses
+from ee.clickhouse.sql.actions import ACTION_QUERY
+from ee.clickhouse.sql.events import GET_EARLIEST_TIMESTAMP_SQL
+from posthog.constants import TREND_FILTER_TYPE_ACTIONS, TRENDS_CUMULATIVE
+from posthog.models.action import Action
+from posthog.models.entity import Entity
 from posthog.models.filter import Filter
 from posthog.models.team import Team
-from posthog.queries.base import BaseQuery
+from posthog.queries.base import BaseQuery, determine_compared_filter
+from posthog.utils import relative_date_parse
+
+# TODO: use timezone from timestamp request and not UTC remove from all below—should be localized to requester timezone
+VOLUME_SQL = """
+SELECT count(*) as total, toDateTime({interval}({timestamp}), 'UTC') as day_start from events where team_id = {team_id} and event = '{event}' {filters} {date_from} {date_to} GROUP BY {interval}({timestamp})
+"""
+
+VOLUME_ACTIONS_SQL = """
+SELECT count(*) as total, toDateTime({interval}({timestamp}), 'UTC') as day_start from events where team_id = {team_id} and id IN ({actions_query}) {filters} {date_from} {date_to} GROUP BY {interval}({timestamp})
+"""
+
+NULL_SQL = """
+SELECT toUInt16(0) AS total, {interval}(now() - number * {seconds_in_interval}) as day_start from numbers({num_intervals})
+"""
+
+AGGREGATE_SQL = """
+SELECT SUM(total), day_start from ({null_sql} UNION ALL {content_sql}) group by day_start order by day_start
+"""
 
 
 class ClickhouseTrends(BaseQuery):
+    def _serialize_entity(self, entity: Entity, filter: Filter, team: Team, label_note: str = "") -> Dict[str, Any]:
+        serialized: Dict[str, Any] = {
+            "action": entity.to_dict(),
+            "label": "{}{}".format("{} — ".format(label_note), entity.name),
+            "count": 0,
+            "data": [],
+            "labels": [],
+            "days": [],
+        }
+
+        # get params
+        inteval_annotation = get_interval_annotation(filter.interval)
+        num_intervals, seconds_in_interval = get_time_diff(filter.interval or "day", filter.date_from, filter.date_to)
+
+        parsed_date_from, parsed_date_to = parse_timestamps(filter=filter)
+
+        prop_filters, prop_filter_params = parse_prop_clauses("id", filter.properties, team)
+
+        # TODO: remove hardcoded params
+        params: Dict = {"team_id": team.pk}
+        params = {**params, **prop_filter_params}
+        if entity.type == TREND_FILTER_TYPE_ACTIONS:
+            action = Action.objects.get(pk=entity.id)
+            action_query, action_params = format_action_filter(action)
```

Comment:
Can use the action relations table but didn't here yet to make sure the queries fully work